HDDS-15427. [Docs] Update Release Manager Guide (#457)
apache/ozone-site: docs/08-developer-guide/04-project/02-release-guide.md

### Write a Haiku

Check the release's national park tag, set in the `<ozone.release>` property of the project's top level pom (also shown by `bin/ozone version`).
Find a photo which is under the CC license.
Write a haiku to the photo with Future font.

### Update the Ozone Website

The website is rendered from the `master` branch of the [apache/ozone-site](https://github.com/apache/ozone-site) repository. Open a pull request against the `master` branch that:

1. Updates the [Downloads page](https://ozone.apache.org/download/) (`src/pages/download.md`) to include the artifacts of the new release version.
2. Adds release notes for the new version under `src/pages/release-notes/` with your haiku image. Please write the release notes in [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) format. In particular, make sure it is human readable and not a simple commit log diff.

> 3. Creates a versioned copy of the documentation for the release. See [this pull request](https://github.com/apache/ozone-site/pull/330) for an example.

Reviewer: ```suggestion
3. Creates a versioned copy of the documentation for the release. See [this pull request](https://github.com/apache/ozone-site/pull/330) for an example, and check out the instructions from the [Docusaurus website](https://docusaurus.io/docs/versioning#tagging-a-new-version).
```

2. Run all steps from the sections [Update the Versions](#update-the-ozone-version-on-the-release-branch) through [Publish a Docker Image for the Release](#publish-a-docker-image-for-the-release), with the following modifications:
    - Do not update the protolock files unless protocol buffers were changed as part of the fix.

>     - When updating the website, replace the superseded release's row on the [Downloads page](https://ozone.apache.org/download/) with this patch version, since we do not want users downloading the original release anymore.

Reviewer: ```suggestion
    - When updating the website, replace the superseded release's row on the [Downloads page](https://ozone.apache.org/download/) with this patch version, since users should no longer download the original release.
```

>       - For example, if 1.2.1 is released, the 1.2.0 row on the Downloads page should be replaced with 1.2.1.

Reviewer: ```suggestion
      - For example, if 1.2.1 is released, replace the 1.2.0 row on the Downloads page with 1.2.1.
```
